src/lazy/system_reader.rs

```diff
@@ -215,9 +250,11 @@ impl<Encoding: Decoder, Input: IonInput> SystemReader<Encoding, Input> {
         if let ExpandedValueRef::List(list) = symbols.read()? {
             for symbol_text_result in list.iter() {
                 if let ExpandedValueRef::String(str_ref) = symbol_text_result?.read()? {
-                    pending_lst.symbols.push(Some(str_ref.text().to_owned()))
+                    pending_lst
+                        .symbols
+                        .push(Symbol::shared(Arc::from(str_ref.deref())))
                 } else {
-                    pending_lst.symbols.push(None)
+                    pending_lst.symbols.push(Symbol::unknown_text())
                 }
             }
         }
```

Comment:
Huh, TIL that non-String types in the `symbols` list should be treated as `null` values. I'd never thought about the case before. From https://amazon-ion.github.io/ion-docs/docs/symbols.html :
> The `symbols` field should be a list of strings. If the field is missing or has any other type, it is treated as if it were an empty list.
> 
> Null elements declare undefined symbol IDs (“gaps”) within the sequence; implementations must handle requests for such symbols the same as if the requested ID beyond the end of the list. Any element of the list that is not a string must be interpreted as if it were null.

Is it worth referring back to the spec where we can, or is that all implicit?